File: src/main/java/com/jackmeng/stl/$Checker.java
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.

package com.jackmeng.stl;

interface $Checker< T >
{
  boolean is_valid(T type);

  default boolean is_all_valid(T[] types)
  {
    boolean e = false;
    for (T x : types)
      e = is_valid(x);
    return e;
  }
}

File: src/main/java/com/jackmeng/stl/$Checker.kt
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.
package com.jackmeng.stl

internal interface `$Checker`<T>
{
	fun is_valid(type:T):Boolean
	fun is_all_valid(types:Array<T>):Boolean
	{
		var e=false
		for (x in types) e=is_valid(x)
		return e
	}
}